packages/baby-jubjub/tests/index.test.ts

```diff
@@ -34,7 +35,7 @@ describe("BabyJubjub", () => {
 
         const expectedPackedPoint = babyjub.packPoint(publicKey)
 
-        expect(packedPoint).toBe(BigInt(`0x${Buffer.from(expectedPackedPoint).toString("hex")}`))
+        expect(packedPoint).toBe(utils.leBuff2int(expectedPackedPoint))
```

Comment:
Optionally, I thought of another good test check which might be worth adding, which would directly reflect the mismatch which caused me to detect this in the first place.

As a bigint, we expect the packed point to be identical to the Y coordinate, except for the 1 bit added to represent the whether the X coordinate is negative or positive.  You could strip off that extra bit and check this expectation like this (syntax may not be perfect, I haven't tried compiling this):

const strippedPackedPoint = packedPoint & ~(1n<<255n);
expect(strippedPackedPoint).toBe(publicKey[0]);

You could also check that the high bit is as-expected based on the value of publicKey[1], but that's unrelated to the byte order issue.

Reply:
Yess, makes sense! I've added it 👍🏽 The coordinate to check is `publicKey[0]` though.

Reply:
Right, I got array indexes wrong in my comment above.  My reference to Y and X was correct.

Side note based on something I learned about today: I often get this confused, and I think it's because the standard form of elliptic curves I've seen in examples is symmetric when reflected across the X axis, not the Y, which doesn't match what's being done here.  I think the important distinction is that the Baby-Jubjub library is using Edwards form of its curve, which seems to be symmetric across both X and Y (https://en.wikipedia.org/wiki/Edwards_curve).  Though it's symmetric on both axes, that article suggests that the definition of the opposite of a point on an Edwards curve by negating X.  That's probably why packPoint is defined the way it is here.

Reply:
>  I think the important distinction is that the Baby-Jubjub library is using Edwards form of its curve, which seems to be symmetric across both X and Y (https://en.wikipedia.org/wiki/Edwards_curve). Though it's symmetric on both axes, that article suggests that the definition of the opposite of a point on an Edwards curve by negating X. That's probably why packPoint is defined the way it is here.

It should be like that. I've discussed this aspect with many people more expert than me and I got a rough idea. This should confirm that idea 🤔